### DISCUSSION TITLE
Does the export command generate cross platform requirements file?

### DISCUSSION THREAD

**Discussion by g-kartik** [pdm-project/pdm#3059] [Q&A] [answered]
I see that that the requirements file generated by the `export` command contain hashes of all available platforms. Does that make it cross platform?

**Comment by g-kartik** [comment]
I generated wheels using the requirements file in a multi-stage docker image. However when trying to install them in the run-time stage using `pip install`, I get the error that the wheel is not a supported wheel on this platform.
For example, I am getting this `cryptography-43.0.0-cp37-abi3-manylinux_2_28_x86_64.whl is not a supported wheel on this platform`

**Comment by frostming** [comment, marked as answer]
> Does that make it cross platform?

It exports what is recorded in the lock file, so if the lock file is cross platform so is the exported file. The hashes come from the lock file,too.

The problem you encountered doesn't seem relevant to this.

**Reply by g-kartik** [reply]
Hello @frostming 
I am satisfied with your answer and I also agree that the problem I encounter is not relavant to pdm. 
For some reason wheel built from the requirements file is not for the correct platform. How this can be? Do you have any idea?

**Reply by g-kartik** [reply]
Hey,
I solved the issue. I was using Ubuntu Focal (20.04). It came with `3.8.10` version of `python` and `20.02` version of `pip`. I thought upgrading to latest version of `pip`, `24.2` is a good idea, however that was the culprit. The latest version did not download the appropriate wheel for the packages. It was not just for `cryptography` but for all packages. Upon downgrading to the default version, the packages are installing fine. It is weird, because latest version of `pip` supports python `3.8`.
